File: src/main/java/com/autosale/shop/repository/impl/AmazonS3ProductClientRepositoryImpl.java
package com.autosale.shop.repository.impl;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.model.AmazonS3Exception;
import com.amazonaws.services.s3.model.S3Object;
import com.autosale.shop.repository.AmazonS3ProductClientRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.io.IOException;
import java.time.LocalDate;

@Repository
@RequiredArgsConstructor
public class AmazonS3ProductClientRepositoryImpl implements AmazonS3ProductClientRepository {

    private final AmazonS3 s3;

    @Override
    public void save(String data) {
        s3.putObject("partsshop", LocalDate.now() + ".csv", data);
    }

    @Override
    public String load(String name) {
        try (S3Object s3Object = s3.getObject("partsshop", name)) {
            return new String(s3Object.getObjectContent().readAllBytes());
        } catch (AmazonS3Exception | IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/com/autosale/shop/repository/impl/AmazonS3ProductClientRepositoryImpl.kt
package com.autosale.shop.repository.impl

import com.amazonaws.services.s3.AmazonS3
import com.amazonaws.services.s3.model.AmazonS3Exception
import com.autosale.shop.repository.AmazonS3ProductClientRepository
import org.springframework.stereotype.Repository
import java.io.IOException
import java.time.LocalDate

@Repository
class AmazonS3ProductClientRepositoryImpl(private val s3: AmazonS3) : AmazonS3ProductClientRepository {

    override fun save(data: String) {
        s3.putObject("partsshop", LocalDate.now().toString() + ".csv", data)
    }

    override fun load(name: String): String {
        try {
            s3.getObject("partsshop", name).use { s3Object -> return String(s3Object.objectContent.readAllBytes()) }
        } catch (e: AmazonS3Exception) {
            throw RuntimeException(e)
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}
